### PR TITLE
[202405][REBASE&FF] Add additional Plugins

### PR DESCRIPTION
**Pull request by Javagedes**
# Preface

Adds the following plugins: `BmpCheckPlugin`, `SamplePlugin`, and `Edk2HelperPlugin`
## Description

Includes the following commits:
3403e333d44859070864612b81b41492077753bd
60c1bc0ef95314b8dac970b87a6221e7549ac555
46c7c4bfa66e7a72fb20b2c5e51e81f50577a2dc
6db656f2bb8a411f06e863b22e60fe6c65341953
4584cb2fa4
3403e333d4

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

N/A

## Integration Instructions

N/A

### CODE REVIEW

File: BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# @file BmpCheck.py
+# Plugin to support checking BMP's included in the FDF for proper usage
+# This makes symbol publishing easier and with the usage of
+# ALT PDB PATH can shrink the size of each module.
+#
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (makubacki) [inline BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py]: For the files in the PR.

```suggestion
# Copyright (c) Microsoft Corporation.
```

File: BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
+# @file BmpCheck.py
+# Plugin to support checking BMP's included in the FDF for proper usage
+# This makes symbol publishing easier and with the usage of
+# ALT PDB PATH can shrink the size of each module.
```
Comment (makubacki) [inline BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py]: This seems like its part of the description for the flatten PDBs plugin.

File: BaseTools/Plugin/BmpCheck/BmpCheck_plug_in.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "scope": "surfacefamily",
```
Comment (makubacki) [inline BaseTools/Plugin/BmpCheck/BmpCheck_plug_in.json:2]: Doesn't have to be done here, but we should consider making this agnostic of surface in the future.

File: BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
+# @file BmpCheck.py
```
Comment (makubacki) [inline BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py]:
```suggestion
# @file BmpCheckPlugin.py
```

File: BaseTools/Plugin/Sample/HelloWorld.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (makubacki) [inline BaseTools/Plugin/Sample/HelloWorld.py]: This is old copyright text that should be able to be removed with the SPDX-License-Identifier present.

**Review by makubacki — APPROVED** [review]

File: BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py (hunk 1 above)
Comment (Javagedes) [reply BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py]: Removed

File: BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py (hunk 4 above)
Comment (Javagedes) [reply BaseTools/Plugin/BmpCheck/BmpCheckPlugin.py]: Updated

File: BaseTools/Plugin/Sample/HelloWorld.py (hunk 5 above)
Comment (Javagedes) [reply BaseTools/Plugin/Sample/HelloWorld.py]: Removed

**Review by os-d — APPROVED** [review]

File: BaseTools/Plugin/BmpCheck/BmpCheck_plug_in.json (hunk 3 above)
Comment (Javagedes) [reply BaseTools/Plugin/BmpCheck/BmpCheck_plug_in.json:2]: I'll knock this out after 202405 is released. The plugin needs an update or two also. Test3 doesn't actually test anything at the moment.

**Pull request merged**